src/rosdistro/manifest_provider/bitbucket.py

```diff
@@ -56,37 +54,29 @@
 BITBUCKET_USER = os.getenv('BITBUCKET_USER', None)
 BITBUCKET_PASSWORD = os.getenv('BITBUCKET_PASSWORD', None)
 
-# Three capture groups are server, org, repo_name. URLs are expected to be well-behaved
-# rosdistro-style URLs either in https or ssh format.
-GIT_REGEX = re.compile('(?:https?:\/\/|ssh:\/\/|git@)([\w.-]+)[:/]([\w-]+)\/([\w-]*)\.git$')
-
 
 def bitbucket_manifest_provider(_dist_name, repo, pkg_name):
     assert repo.version
+    server, path = repo.get_url_parts()
 
-    match = GIT_REGEX.match(repo.url)
-    if not match:
-        raise RuntimeError('Unexpected git URL format: %s' % repo.url)
-
-    server, org, repo_name = match.groups()
     if server != 'bitbucket.org':
         logger.debug('Skip non-bitbucket url "%s"' % repo.url)
         raise RuntimeError('Cannot handle non bitbucket url.')
 
-    release_tag = get_release_tag(repo, pkg_name)
+    release_tag = repo.get_release_tag(pkg_name)
 
-    if not check_remote_tag_exists(repo.url, release_tag):
+    if release_tag not in repo.remote_tags:
         raise RuntimeError('specified tag "%s" is not a git tag' % release_tag)
 
-    url = 'https://%s/%s/%s/raw/%s/package.xml' % (server, org, repo_name, release_tag)
+    url = 'https://bitbucket.org/%s/raw/%s/package.xml' % (path, release_tag)
     try:
         logger.debug('Load package.xml file from url "%s"' % url)
         req = Request(url)
         if BITBUCKET_USER and BITBUCKET_PASSWORD:
             logger.debug('- using http basic auth from supplied environment variables.')
-            authheader = 'Basic %s' % base64.encodestring('%s:%s' % (BITBUCKET_USER, BITBUCKET_PASSWORD))
+            authheader = 'Basic %s' % base64.b64encode('%s:%s' % (BITBUCKET_USER, BITBUCKET_PASSWORD))
             req.add_header('Authorization', authheader)
-        package_xml = urlopen(req).read()
+        package_xml = urlopen(req).read().decode('utf-8')
```

Comment:
Without this change, the new unit tests fail under Python 3 with the error `Type str doesn't support the buffer API`. Full conversion of the cache to native unicode is in #80.